src/NuGet.Core/NuGet.Protocol/Plugins/RequestHandlers/GetCredentialsRequestHandler.cs

```diff
@@ -154,19 +155,29 @@ public void AddOrUpdateSourceRepository(SourceRepository sourceRepository)
                         cancellationToken);
                 }
 
-                if (credential == null)
+                if (credential is AuthTypeFilteredCredentials filteredCredentials)
```

Comment:
Actually this would throw in the case of null. 

An equivalent example would be the following: 
```
using System;

public class Program
{
    public static void Main()
    {
        object[] items = { "bla", null };
        foreach (var item in items)
        {
            if (item is var obj)
                Console.WriteLine($"Type: {obj.GetType().Name}, Value: {obj}");
        }
    }
}
```